plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -151,18 +158,68 @@ func (h *HTTPListenerV2) Init() error {
 		return err
 	}
 
+	protoRegex := regexp.MustCompile(`\w://`)
+	if !protoRegex.MatchString(h.ServiceAddress) {
+		h.ServiceAddress = "tcp://" + h.ServiceAddress
+	}
+
+	var u *url.URL
+	// Parse and check the address
+	if runtime.GOOS == "windows" && strings.HasPrefix(h.ServiceAddress, "unix://") {
+		u = &url.URL{Scheme: "unix", Path: strings.TrimPrefix(h.ServiceAddress, "unix://")}
+	} else {
```

Comment:
Is this really necessary? Parsing a unix URL should also work on Windows, shouldn't it?

Reply:
Yes it was problematic as sample address will look like `unix://c:/Users/bazko1/temp/telegraf.socket` the `url.Parse` starts to interpret the part after `c:` as port and it results in an parse error.

You can check that in `socket_listener` tests for windows are skipped with comment that the unixgram is not supported even though the unix socket is.
See: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/socket_listener/socket_listener_test.go#L125-L128

Reply:
I think the correct way is to use `unix:///C:/Users/bazko1/...` (i.e. three slashes) according to https://github.com/golang/go/commit/844b625ebcc7101e09fb87828a0e71db942a2416...

Reply:
Yes great I tested it and this works I will remove the windows case then.